azure_li_services/units/system_setup.py

```diff
@@ -40,7 +43,7 @@ def main():
         set_hostname(hostname)
 
     set_kdump_service(
-        config.get_crash_kernel_high(), config.get_crash_kernel_low()
+        config.get_crash_kernel_high(), config.get_crash_kernel_low(), status
```

Comment:
Not quite clear here. The desired affect here, if we can have it, should be:

1. Get recommended values of crash_high and crash_low for this system,
1. If overridden in YAML, use overrides. Otherwise use recommended values.
1. If values are what we booted with (no changes), do not reboot system. Otherwise, reboot to take effect.

It's not clear to me that this is what the code is reflecting. Is it?

Reply:
This is exactly how I implemented it yes :) However if the code flow has confused you and lead to the questions of how the logic works we should come up with better coding. So do you have a certain place in code which was not clear or confusing ? I'm asking because the review point you wrote the message for is just pointing to a function call

Reply:
The function call is what made me think of it, don't use that as a reflection of the logic of the code! 😄 

As I looked over the changes, I expected a segment of code along the lines of:

1. Check system-recommended settings for high, low, save them.
1. If Yaml(high) != saved_high || Yaml(low) != saved_low { saved_high = Yaml(high); saved_low = Yaml(low) }
1. If saved_high != current_system_high || saved_low != current_system_low { trigger kexec reboot }

So I expected that sort of code segment, perhaps split across several areas of code, but in looking around I didn't see those segments in any obvious place. Now, of course, you might have the segments hidden in assignment operators and such, which would help me to miss it.

Not seeing those segments, I wondered about the algorithm, so I asked. But if the algorithm is what I intended, I'm satisfied. I'll sign off on this, but if you choose to clarify code layout, you know what I was expecting/looking for.

Reply:
You are only seeing the diff, the logic you are looking for is in the function _kdump_calibrate in units/system_setup.py was reviewed and merged a while ago with c1ff0e33

Reply:
correct, if you checkout the branch instead of just looking on the diff you will see the complete logic which I think is a good read to understand the logic.

If you still see room for refactoring from a readability point of view feel free to open an issue.

Thanks